### DISCUSSION TITLE
macOS nightly: move to ARM builds?

### DISCUSSION THREAD

**Discussion by iandol** [jgm/pandoc#10374] [General]
Currently, the pandoc nightlies are built for older Intel machines, and while Apple Silicon machines can run Intel code via emulation, there are problems like using Lua packages in filters. To switch to Apple Silicon, the following line needs to change from `macos-13` to `macos-14` or `macos-latest`:

https://github.com/jgm/pandoc/blob/main/.github/workflows/nightly.yml#L72

https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/choosing-the-runner-for-a-job#standard-github-hosted-runners-for-public-repositories

Does anyone have a preference for which system to build for?

**Comment by jgm** [comment]
>  the following line needs to change from macos-13 to macos-14 or macos-latest

That would be sufficient?  Are the macos-14 and later build machines ARM machines?

Personally I suspect that the majority of macos users may already be on apple silicon.  However, the point of nightlies is not to provide people with efficient pandoc binaries.  We don't really want to encourage people to run non-released versions.  The purpose is to allow people to test, mainly, or to workaround bugs that have been fixed that affect their workflows.  A slightly slower executable is okay for these purposes.  And if we compile for apple silicon, and intel mac users can't use the nightly, that interferes with these purposes.

**Reply by iandol** [reply]
The issue i had was that I wanted to test the recent fixes you made for docx+citations outputs while debugging a Lua filter that will convert the ids to use BibTeX citation keys and the nightly fails to work with the Lua debugger I use as the intel build can't use ARM libraries. I am certainly an edge case in this regard...

You are right that probably few people need to use the nightlies, and what percentage use intel vs. arm is unknown...

> That would be sufficient?

From the Github docs that is what I understood, macos-14 and later all run on Apple Silicon machines only.
